binr/rabin2/rabin2.c

```diff
@@ -899,18 +899,9 @@ int main(int argc, char **argv) {
 			return 1;
 		}
 		codelen = r_hex_str2bin (p, code);
-		if (!arch) {
-			arch = R_SYS_ARCH;
-		}
-		if (!bits) {
-			bits = 32;
-		}
-		if (!r_bin_use_arch (bin, arch, bits, create)) {
-			eprintf ("Cannot set arch\n");
-			r_core_fini (&core);
-			return 1;
-		}
-		b = r_bin_create (bin, code, codelen, data, datalen);
+		RBinArchOptions opts;
+		r_bin_arch_options_init (&opts, arch, bits);
+		b = r_bin_create (bin, create, code, codelen, data, datalen, &opts);
```

Comment:
those options of arch+bits is something that can live inside RBin, like it does in RDebug, RAsm, RAnal, RSyscall, .... iknow this stuff is only going to be used in RBinCreate and in RBinObjects and should be separate instances.. but im just trying to throw some comments to make this more consistent